avo-hq/avo
# fix: actions and field hydration

# Description


Fixes #1946

This pull request addresses primarily two key issues: the class attributes for actions and the field hydration method.

## Action class attributes
Relying on class attributes for actions is not convenient since undesired old values can remain populated on a new instance.

## Field Hydration Method
The existing code made it impossible to hydrate the field with `nil` values, resulting in undesired behavior. For instance, in cases where a field within an action relied on the `@record` via a default Proc, it would correctly receive the default value when executed in a show view. However, when executed in an index view, the field attempted to hydrate with `model: nil`, yet the prior `model` would remain hydrated, causing the value to be fetched incorrectly.

## `ActionModel` removed
`ActionModel` was not being used only to pass it as `model` on `form_with` helper, and I think that was unecessary.



# Checklist:




## Manual review steps


1. Create an action with several fields that have the `default` value as Proc reling on the `@record`
  Example: 
```ruby
class ReleaseFish < Avo::BaseAction
  field :option_a, as: :boolean, default: -> { @record&.foo? }
  field :option_b, as: :boolean, default: -> { @record&.foo? }
  field :option_c, as: :boolean, default: -> { @record&.foo? }

  def handle(...)
    succeed ":)"
  end
end
```
2. Define the method on the model
  Example:
```ruby
  def foo?
    true
  end
```
3. Register the action on a resource
4. Go to that resource index page
5. Select 2 records
6. Run the action
7. Notice that all fields are false since `@record` is blank
8. Go to one record show view
9. Run the action
10. Notice that all fields are true since `@record` is setted ( and the default proc is running with `@record` properly hydrated on all executions )
11. Return to index and run action after selecting 2 records
12. Notice the same as step 7 (Prior this commit the fields would still hydrated with the `@record` from the show view since was impossible to hydrate with nil values)

Manual reviewer: please leave a comment with output from the test if that's the case.
